airbyte-webapp/src/views/Connection/ConnectionForm/formConfig.tsx

```diff
@@ -90,8 +90,109 @@ export const createConnectionValidationSchema = ({
   mode,
   allowSubOneHourCronExpressions,
   allowAutoDetectSchema,
-}: CreateConnectionValidationSchemaArgs) =>
-  yup
+}: CreateConnectionValidationSchemaArgs) => {
+  const isNewStreamsTableEnabled = process.env.REACT_APP_NEW_STREAMS_TABLE ?? false;
+
+  if (isNewStreamsTableEnabled) {
```

Comment:
How come this is duplicated, what is the different between the two?

Reply:
There are different approaches to validating the form 

The new implementation (we have independent rules for each config value):
![image](https://user-images.githubusercontent.com/66960359/210436606-1470a124-85e2-4228-b491-5980594b960f.png)

Old implementation (with this implementation, we can't understand where is an error in the config):
![image](https://user-images.githubusercontent.com/66960359/210436740-d3051e54-e47c-4e56-8679-ced7fdd06fa9.png)

Reply:
Got it. One way to avoid this duplication is by declaring config first based on which schema we are using, then assigning to the object that is returned

Reply:
Fixed by a conditional operator inside the yup object